hoomd/md/minimize/fire.py

```diff
@@ -0,0 +1,198 @@
+# coding: utf-8
+
+# Copyright (c) 2009-2021 The Regents of the University of Michigan
+# This file is part of the HOOMD-blue project, released under the BSD 3-Clause
+# License.
+
+# Maintainer: joaander / All Developers are free to add commands for new
+# features
+
+"""Energy minimizer."""
+
+import hoomd
+
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.data.typeconverter import OnlyFrom, OnlyTypes
+from hoomd.md import _md
+from hoomd.md.integrate import _DynamicIntegrator, _preprocess_aniso
+
+
+def _raise_if_nonpositive(value):
+    if value <= 0:
+        raise ValueError("This value cannot be negative")
+    return value
+
+
+class FIRE(_DynamicIntegrator):
+    """Energy Minimizer (FIRE).
+
+    Args:
+        dt (float):
+            This is the maximum step size the minimizer is permitted to use.
+            Consider the stability of the system when setting.
+            math:`[\\mathrm{time}]`
+        min_steps_adapt (int):
+            Number of steps energy change is negative before allowing
+            :math:`\\alpha` and :math:`\\delta t` to adapt.
+        finc_dt (float):
+            Factor to increase :math:`\\delta t` by
+        fdec_dt (float):
+            Factor to decrease :math:`\\delta t` by
+        alpha_start (float):
+            Initial (and maximum) :math:`\\alpha`
+        fdec_alpha (float):
+            Factor to decrease :math:`\\alpha t` by
+        force_tol (float):
+            force convergence criteria
+            math:`[\\mathrm{force} / \\mathrm{mass}]`
+        angmom_tol (float):
+            angular momentum convergence criteria
+            math:`[\\mathrm{energy} * \\mathrm{time}]`
+        energy_tol (float):
+            energy convergence criteria math:`[\\mathrm{energy}]`
+        min_steps_conv (int):
+            A minimum number of attempts before convergence criteria are
+            considered
+        aniso (bool):
+            Whether to integrate rotational degrees of freedom (bool), default
+            None (autodetect).
+
+    Added in version 2.2
+
+    .. versionadded:: 2.1
+    .. versionchanged:: 2.2
```

Comment:
We have been removing these in the conversion to version 3 since this notice is not meaningful in version 3.